packages/components/src/QueryModel/GridPanel.tsx

```diff
@@ -143,11 +150,12 @@ interface State {
     actionValues: ActionValue[];
 }
 
-export class GridPanel extends PureComponent<Props, State> {
+export class GridPanel<T> extends PureComponent<Props<T>, State> {
```

Comment:
nit: As discussed, could consider declaring the `GridPanel` class with a default value of `{}` for `T`:

```ts
export class GridPanel<T = {}> extends PureComponent<Props<T>, State>
```

Then, if this is working as I'd expect you can remove the declarations of `GridPanel<{}>` in the tests.